File: php-laraval-app/app/Models/Team.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Team extends Model
{
    use HasFactory;
    use SoftDeletes;

    /**
     * Indicates if the model should be timestamped.
     *
     * @var bool
     */
    public $timestamps = true;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'name',
        'avatar',
        'color',
        'description',
        'is_active',
    ];

    /**
     * The accessors to append to the model's array form.
     *
     * @var array
     */
    protected $appends = [
        'total_workers',
    ];

    /**
     * The attributes that should be cast to native types.
     *
     * @var array
     */
    protected $casts = [
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
        'deleted_at' => 'datetime',
        'is_active' => 'boolean',
    ];

    /**
     * Define the relationships that an accessor uses.
     *
     * @var array<string,array<string,string[]>>
     */
    protected array $accessorsRelations = [
        'total_workers' => ['users' => ['id']],
    ];

    public function getTotalWorkersAttribute()
    {
        return $this->users->count();
    }

    public function users(): BelongsToMany
    {
        return $this->belongsToMany(User::class);
    }

    public function subscriptions(): HasMany
    {
        return $this->hasMany(Subscription::class);
    }

    public function scheduleCleanings(): HasMany
    {
        return $this->hasMany(ScheduleCleaning::class);
    }

    public function schedules(): HasMany
    {
        return $this->hasMany(Schedule::class);
    }
}

File: dotnet/src/Downstairs.Infrastructure/Persistence/Models/Team.cs
namespace Downstairs.Infrastructure.Persistence.Models;

public partial class Team
{
    public long Id { get; set; }

    public string Name { get; set; } = null!;

    public string? Avatar { get; set; }

    public string? Color { get; set; }

    public string? Description { get; set; }

    public bool? IsActive { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public DateTime? DeletedAt { get; set; }

    public virtual ICollection<LaundryOrder> LaundryOrderDeliveryTeams { get; set; } = new List<LaundryOrder>();

    public virtual ICollection<LaundryOrder> LaundryOrderPickupTeams { get; set; } = new List<LaundryOrder>();

    public virtual ICollection<ScheduleCleaning> ScheduleCleanings { get; set; } = new List<ScheduleCleaning>();

    public virtual ICollection<Schedule> Schedules { get; set; } = new List<Schedule>();

    public virtual ICollection<SubscriptionCleaningDetail> SubscriptionCleaningDetails { get; set; } = new List<SubscriptionCleaningDetail>();

    public virtual ICollection<SubscriptionLaundryDetail> SubscriptionLaundryDetailDeliveryTeams { get; set; } = new List<SubscriptionLaundryDetail>();

    public virtual ICollection<SubscriptionLaundryDetail> SubscriptionLaundryDetailPickupTeams { get; set; } = new List<SubscriptionLaundryDetail>();

    public virtual ICollection<Subscription> Subscriptions { get; set; } = new List<Subscription>();

    public virtual ICollection<TeamUser> TeamUsers { get; set; } = new List<TeamUser>();
}